apps/linode-marketplace-beef/roles/beef/tasks/main.yml

```diff
@@ -0,0 +1,175 @@
+---
+- name: install required packages
+  apt:
+    pkg:
+      - nginx
+      - curl
+      - git
+      - build-essential
+      - openssl
+      - libreadline6-dev
+      - zlib1g
+      - zlib1g-dev
+      - libssl-dev
+      - libyaml-dev
+      - libsqlite3-0
+      - libsqlite3-dev
+      - sqlite3
+      - libxml2-dev 
+      - libxslt1-dev
+      - autoconf
+      - libc6-dev
+      - libncurses5-dev
+      - automake
+      - libtool
+      - bison
+      - nodejs
+      - libcurl4-openssl-dev
+      - ruby-dev
+    state: latest
+    update_cache: true
+
+# Nginx
+- name: config file setup
+  template:
+      src: 'templates/nginx.conf.j2'
+      dest: '/etc/nginx/sites-available/{{ _domain }}'
+
+- name: create certs dir
+  file:
+    path: '/var/www/certs/.well-known'
+    state: directory
+    recurse: yes
+
+- name: changing owner on certs dir
+  file:
+    path: '/var/www/certs/'
+    state: directory
+    recurse: yes
+    owner: www-data
+    group: www-data
+
+- name: create a symbolic link for the nginx configuration
+  file:
+    src: '/etc/nginx/sites-available/{{ _domain }}'
+    dest: '/etc/nginx/sites-enabled/{{ _domain }}'
+    state: link
+
+- name: unlink the default nginx configuration
+  file:
+    path: '/etc/nginx/sites-enabled/default'
+    state: absent
+  notify: reload nginx service
+
+# ssl_certbot
+- name: setting up ssl certificate
+  import_role:
+    name: certbot_ssl
+
+# beef setup    
+
+- name: create beef group
+  group:
+    name: beef
+    state: present
+
+- name: create beef user
+  user:
+    name: beef
+    shell: /sbin/nologin
+    groups: beef
+    system: true
+    create_home: false
+
+- name: clone beef repo
+  git:
+    repo: https://github.com/beefproject/beef.git
+    dest: '/home/beef'
+
+- name: changing beef dir owner
+  file:
+    path: /home/beef
+    owner: beef
+    group: beef
+    recurse: true
+
+- name: copying lets encrypt certs to beef dir
+  copy:
+    src: '{{ item.src }}' 
+    dest: '{{ item.dest }}'
+  loop:
+    - src: '/etc/letsencrypt/live/{{ _domain }}/privkey.pem'
+      dest: '/home/beef'
+    - src: '/etc/letsencrypt/live/{{ _domain }}/fullchain.pem'
+      dest: '/home/beef'
+
+- name: adding config file
+  template: 
+    src: 'beef_config.yaml.j2'
+    dest: '/home/beef/config.yaml'
+
+- name: bundle config setup
+  command: bundle3.0 config set --local path /home/beef/.gem
+  become_user: beef
+  become: true
+  args:
+   chdir: /home/beef
+
+- name: Install dependencies using bundle
+  command: bundle3.0 install
+  become_user: beef
+  become: true
+  args:
+   chdir: /home/beef
```

Comment:
we could use the `community.general.bundler` module for this:
https://docs.ansible.com/ansible/latest/collections/community/general/bundler_module.html